coralproject/talk
# [CORL-2528]: site moderators can unban on their sites

## What does this PR do?
These changes make it so that site moderators are able to unban users on their site(s).

I also migrated over all community tests related to site banning. Since that was a large test file, I broke these out into their own `banUser.spec.tsx` file.

## What changes to the GraphQL/Database Schema does this PR introduce?
none

## Does this PR introduce any new environment variables or feature flags? 
no

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?

Signed in as a site moderator, go to the `Community` page in the admin. Ban a user across a site within that site moderator's scope. Then click the user status dropdown again and see that you are able to select `Manage ban` for that user. Unban them on the site they were banned on.

Also test that when a user is banned across a few different sites, some within and some outside of a site moderator's scope, the site moderator can only unban on any sites that are within their scope. Other sites should have their checkboxes disabled.

Also test that other banning/unbanning functionality is still as expected, for site moderators and admins.

Test that site moderators cannot unban their site(s) if there is an all-site ban in place. This can only be undone by an admin or org moderator.
 
## How do we deploy this PR?
